app/src/main/java/org/mozilla/fenix/home/HomeMenu.kt

```diff
@@ -30,19 +32,56 @@ class HomeMenu(
         object History : Item()
         object Bookmarks : Item()
         object Quit : Item()
+        object Sync : Item()
     }
 
     val menuBuilder by lazy { BrowserMenuBuilder(menuItems) }
 
     private val hasAccountProblem get() = context.components.backgroundServices.accountManager.accountNeedsReauth()
     private val primaryTextColor =
         ThemeManager.resolveAttribute(R.attr.primaryText, context)
+    private val syncDisconnectedColor = ThemeManager.resolveAttribute(R.attr.syncDisconnected, context)
+    private val syncDisconnectedBackgroundColor = context.getColorFromAttr(R.attr.syncDisconnectedBackground)
 
     private val menuCategoryTextColor =
         ThemeManager.resolveAttribute(R.attr.menuCategoryText, context)
 
     private val menuItems by lazy {
-        val items = mutableListOf(
+
+        val items = mutableListOf<BrowserMenuItem>()
+
+        if (hasAccountProblem) {
+            items.add(
```

Comment:
Optional change here, but I personally find this list a little bit difficult to read.  There's some similar code in `DefaultToolbarMenu` that I changed to look like the below, which I personally find easier to understand:

```kotlin
        val menuItems = listOfNotNull(
            help,
            settings,
            library,
            desktopMode,
            addToTopSites,
            addToHomescreen.apply { visible = ::shouldShowAddToHomescreen },
            addons,
            findInPage,
            privateTab,
            newTab,
            if (shouldShowWebcompatReporter) reportIssue else null,
            if (shouldShowSaveToCollection) saveToCollection else null,
            if (shouldDeleteDataOnQuit) deleteDataOnQuit else null,
            readerMode.apply { visible = ::shouldShowReaderMode },
            readerAppearance.apply { visible = ::shouldShowReaderAppearance },
            openInApp.apply { visible = ::shouldShowOpenInApp },
            BrowserMenuDivider(),
            menuToolbar
        )
```